### PR TITLE
Simplify <an+b> in selector args when serializing.

### LINKED ISSUE #17182: Simplify `an+b` when serializing
When we serialize an `an+b` value, we just print it as `an+b`.

We should be simplifying it as much as possible. This means:

 - Printing out only `b` if `a` is zero
 - Printing out only `an` if `b` is zero
 - Printing `0` if both are zero
 - Not printing `a` if it is 1

Reduced testcase:

```html
<style type="text/css">
  :nth-child(+1n+5){}
</style>
<script type="text/javascript">document.write(document.styleSheets[0].cssRules[0].selectorText</script>
```

(Same for `+1n+0`, `+0n+5`, `+0n-0`, etc)

Relevant code: https://github.com/servo/servo/blob/2da8eb0342f6701b5365116b96113c83a1599b1f/components/selectors/parser.rs#L743

Testing: `tests/wpt/web-platform-tests/cssom/selectorSerialize.html`

### PR DESCRIPTION
**Pull request by jyc**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes is part of a series to fix #17182 

I am running this to identify what tests will fail; as the other PRs in the series are merged, I believe the tests that pass will change as well.

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17539)

### CODE REVIEW

**Comment by emilio** [comment]
@bors-servo try

Is this ready to land @jyc?

**Comment by jyc** [comment]
Hi Emilio! I was planning on waiting until the other PRs in this series landed, because then this change should cause some tests to pass. I will go check on those now!

**Comment by jyc** [comment]
At last!

@bors-servo retry

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /cssom-1_dev/html/selectorSerialize.htm:
  └ PASS [expected FAIL] :nth-child serialization produces canonical form

  ▶ Unexpected subtest result in /cssom-1_dev/html/selectorSerialize.htm:
  └ PASS [expected FAIL] single pseudo (simple) selector "nth-child" which accepts arguments in the sequence of simple selectors that is not a universal selector

  ▶ Unexpected subtest result in /cssom-1_dev/html/selectorSerialize.htm:
  └ PASS [expected FAIL] single pseudo (simple) selector "nth-last-child" which accepts arguments in the sequence of simple selectors that is not a universal selector

  ▶ Unexpected subtest result in /cssom-1_dev/html/selectorSerialize.htm:
  └ PASS [expected FAIL] single pseudo (simple) selector "nth-of-child" which accepts arguments in the sequence of simple selectors that is not a universal selector

  ▶ Unexpected subtest result in /cssom-1_dev/html/selectorSerialize.htm:
  └ PASS [expected FAIL] single pseudo (simple) selector ":nth-last-of-type" which accepts arguments in the sequence of simple selectors that is not a universal selector
```

File: components/selectors/parser.rs
Comment (emilio) [inline components/selectors/parser.rs]: nit: I think the name of the function has an unicode character? Could you convert that in two `f`s?

**Review by emilio — APPROVED** [review]

File: components/selectors/parser.rs
Comment (emilio) [inline components/selectors/parser.rs]: Err, this is actually a gh issue, or something like that.

**Pull request merged**
